sample-apps/discount-functions-sample-app/web/frontend/vite.config.js

```diff
@@ -1,32 +1,55 @@
-import { defineConfig, loadEnv } from 'vite';
+import { defineConfig } from "vite";
+import { dirname } from "path";
+import { fileURLToPath } from "url";
+import react from '@vitejs/plugin-react';
 import path from 'path';
+if (
+  process.env.npm_lifecycle_event === "build" &&
+  !process.env.CI &&
+  !process.env.SHOPIFY_API_KEY
+) {
+  console.warn(
+    "\nBuilding the frontend app without an API key. The frontend build will not run without an API key. Set the SHOPIFY_API_KEY environment variable when running the build command.\n"
+  );
+}
 
-// prettier-ignore
-const INDEX_ROUTE = "^/(\\?.*)?$";
-const API_ROUTE = '^/api/';
-const ENV_KEYS = [
-  'SHOPIFY_API_KEY',
-  'SHOPIFY_ORDER_DISCOUNT_ID',
-  'SHOPIFY_SHIPPING_DISCOUNT_ID',
-  'SHOPIFY_PRODUCT_DISCOUNT_ID',
-];
+const proxyOptions = {
+  target: `http://127.0.0.1:${process.env.BACKEND_PORT}`,
+  changeOrigin: false,
+  secure: true,
+  ws: false,
+};
 
-const root = new URL('.', import.meta.url).pathname;
+const host = process.env.HOST
+  ? process.env.HOST.replace(/https:\/\//, "")
+  : "localhost";
+
+// HMR doesn't work on Firefox using localhost, so you can temporarily get that to work by setting the
+// SHOPIFY_VITE_HMR_USE_POLLING env var when running this
+let hmrConfig;
 
-// Function IDs are populated in ../../.env by the Shopify CLI after being deployed
-if (process.env.npm_lifecycle_event === 'dev') {
-  const envFile = loadEnv('dev', path.join(process.cwd(), '..', '..'), '');
-  process.env = { ...process.env, ...envFile };
+if (host === "localhost") {
```

Comment:
macOS uses a `[computer name].local` domain — sometimes as an alias for localhost and more commonly for locally networked machines. I've seen code in the wild fail on macOS when it doesn't check both. Not sure if that applies / matters here but I got into the habit of checking for `.local` domains alongside `localhost`.

<img width="359" alt="macOS local URLs" src="https://user-images.githubusercontent.com/4842862/186547682-bd8a5f2c-5f23-421a-8f63-c0a9dba2e4d2.png">

Reply:
If I understand correctly, the localhost here, refers to where ngrok forwards traffic.

Reply:
Ahh, that shouldn't be a problem then. Thanks for clarifying :)